sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitDecl.cs

```diff
@@ -293,6 +294,13 @@ private void VisitEnumConstantDecl(EnumConstantDecl enumConstantDecl)
             parentName = _outputBuilder.Name;
         }
 
+        if (Config.StripEnumMemberTypeName)
+        {
+            Regex stripParentNameRegex = new($"^{Regex.Escape(parentName)}_*", RegexOptions.IgnoreCase);
+            var strippedName = stripParentNameRegex.Replace(escapedName, string.Empty);
+            escapedName = strippedName;
+        }
```

Comment:
I don't think this needs to be/should be a regex. We're really not doing anything fancy here.

This can really just be like `EscapeAndStripName` is doing, where we're just doing:
```csharp
private string EscapeAndStripEnumConstantName(string name)
{
    if (name.StartsWith(parentName, StringComparison.Ordinal))
    {
        var startIndex = parentName.Length;

        if ((name.Length > startIndex) && (name[startIndex] == '_'))
        {
            startIndex += 1;
        }

        name = name[startIndex..];
    }

    return EscapeName(name);
}
```

-- `EscapeAndStripName` should likely be renamed to `EscapeAndStripMethodName`, a similar change should be made to `PrefixAndStripName` since it also only applies to methods today.

Reply:
@tannergooding I have created a new general-purpose method `PrefixAndStrip` and made the enum stripping logical call that. I have also made `EscapeAndStripName` and `PrefixAndStripName` call that new method as well and renamed them as you suggested.

See commit https://github.com/dotnet/ClangSharp/pull/527/commits/6703d6a4cbbcd82557057c6443724bd929b0ba05